### DISCUSSION TITLE
Bootstrapper setup - digital signing.

### DISCUSSION THREAD

**Discussion by saiprabhur** [oleg-shilo/wixsharp#1633] [Q&A]
we were using paid version digital signature to sign our executable.
the digital signature was working if i am doing sign the MSI.

When we do digital signature boot strapper then setup fails

setup installation error
Log file error:
E574:10494][2024-09-10T19:46:12]e000: Error 0x80070001: Failed to extract all files from container, erf: 1:2:0
[E574:DD94][2024-09-10T19:46:12]e000: Error 0x80070001: Failed to wait for operation complete.
[E574:DD94][2024-09-10T19:46:12]e000: Error 0x80070001: Failed to open container.
[E574:DD94][2024-09-10T19:46:12]e000: Error 0x80070001: Failed to open container: WixAttachedContainer.
[E574:DD94][2024-09-10T19:46:12]e312: Failed to extract payloads from container: WixAttachedContainer to working path: C:\Users\srajendr\AppData\Local\Temp\{0B8FD832-E506-44B5-87B6-1ECBE753D32B}\80E86D31981DEB54B96512E754E9E5DBFFB40CAD, error: 0x80070001.

I am using NineDigit.WixSharpExtensions for bootstrapper.
Wixsharp version: 1.25.3 and NineDigit.WixSharpExtensions version:1.0.14

Could you please help me on this?  Why its failing in the boot strapper.

**Comment by saiprabhur** [comment]
bootstrapper.DigitalSignature = new DigitalSignatureBootstrapper()
  {
      Description = digitalsignaturetest,
      TimeUrl = new Uri(CertificateTimestampUrl),
      OptionalArguments = $"/a /v /s My /sha1 {Thumbprint} /fd sha256"
  };

in the boot strapper i have used the code above to sign the setup

Wixtoolset forum they were describing we have to special care to sign bundle exe.
https://wixtoolset.org/docs/v3/overview/insignia/

do we have to do anything special ?

**Reply by oleg-shilo** [reply]
Nothing that I am aware of.
But keep in mind this https://github.com/oleg-shilo/wixsharp/discussions/1628#discussioncomment-10591374 ...
